feature: blog april (#71)
mun-lang/website: content/posts/2021-05-08-this-month-april.md

---
title: "This Month in Mun - April 2021"
author: "The Mun Team"

> excerpt: "After the release of Mun v0.3 we are now looking at new features for the Mun v0.4 release. The biggest new features will probably be support for arrays."

Reviewer: ```suggestion
excerpt: "After the release of Mun v0.3 we are now looking at new features for the Mun v0.4 release. The biggest of which will be support for dynamically sized arrays."
```

---

After the [release of Mun v0.3][v3r] we are now looking at new features for the Mun v0.4 release. 

> The biggest new features will be support for arrays.

Reviewer: ```suggestion
The biggest of which will be support for dynamically sized arrays.
```

> #### Mun v0.4

Reviewer: I'd add a small section with link about the Mun v0.3 release - as a reminder for people. I think we also did that in the monthly blog post after the Mun v0.2 release.

For reference, https://mun-lang.org/blog/2020/05/31/this-month-may/

Author: Good idea! I added a little section!

Mun v0.4 will be the next release of the Mun programming language. 
We expected to release Mun v0.3 within a few months of Mun v0.2 but this was pushed back a couple of times because we added more and more features to the release. 
We concluded that the next time around we will try to create smaller increments with less features so we are (hopefully) able to release faster and more often. As it stands the [Mun v0.4 milestone][v4m] is a lot smaller than our original Mun v0.3 milestone.

* **[RFC] dynamically sized arrays** [[PR#324]](https://github.com/mun-lang/mun/pull/324)

  The next big feature for Mun will be support for arrays.
 
  To gather feedback and get everyone on the same page we opened an RFC pull request where we encourage everyone to comment on our design choices with regards to array support.

>   We hope to merge this RFC soon so we can get to work on the actual development process of this long awaited feature.

Reviewer: ```suggestion
  We hope to merge this RFC soon so we can get to work on the actual development process of this long-awaited feature.
```

Interested in our progress or what we are doing next?

> Please check out our [milestone page](https://github.com/mun-lang/mun/milestones) on Github that details new features for future releases - Mun v0.4, and v0.5 - as well as a backlog of features that are still to come.

Reviewer: ```suggestion
Please check out our [milestone page](https://github.com/mun-lang/mun/milestones) on Github that details new features for future releases - Mun v0.4 and v0.5 - as well as a backlog of features that are still to come.
```

#### Community

If you are interested in helping develop Mun - but are not sure where to start - feel free to take a look at our [*good first issues*][gfi] or reach out to us on [Discord](https://discord.gg/SfvvcCU) or [Twitter][twi]. To support our cause, please consider donating to our [Open Collective][oc] or [Github Sponsors][gs].

> [doh]: https://hacktoberfest.digitalocean.com

Reviewer: ```suggestion
```

---
title: "This Month in Mun - April 2021"
author: "The Mun Team"
excerpt: "After the release of Mun v0.3 we are now looking at new features for the Mun v0.4 release. The biggest of which will be support for dynamically sized arrays."
---

After the [release of Mun v0.3][v3r] we are now looking at new features for the Mun v0.4 release. 
The biggest of which will be support for dynamically sized arrays.

#### Mun v0.3

> Last month we were able to release Mun v0.3. 

Reviewer: ```suggestion
This month was highlighted by the release of Mun v0.3. 
```

> This release has take us quite some time but we finally managed to put the finishing touches to the third release of Mun. 

Reviewer: ```suggestion
We expected to release Mun v0.3 within a few months of Mun v0.2 but this was pushed back a couple of times because we added more and more features to the release.
```

Mun v0.3 adds support for modules & multi-file projects, a functioning language server which features autocompletion and a lot more. 

> The [Mun v0.3 release blog](https://mun-lang.org/blog/2021/04/11/release-mun-v0-3-0/) highlights all of the new features packed in this release.

Reviewer: ```suggestion
The [Mun v0.3 release blog][v3r] highlights all of the new features packed in this release.
```

<img src="../images/autocomplete.gif" alt="Autocomplete working for Mun in VSCode " />

#### Mun v0.4

Mun v0.4 will be the next release of the Mun programming language. 

> We expected to release Mun v0.3 within a few months of Mun v0.2 but this was pushed back a couple of times because we added more and more features to the release. 
> We concluded that the next time around we will try to create smaller increments with less features so we are (hopefully) able to release faster and more often. As it stands the [Mun v0.4 milestone][v4m] is a lot smaller than our original Mun v0.3 milestone.

Reviewer: ```suggestion
After Mun v0.3's delays, we concluded that from now on we will create smaller increments - with less features - so we are (hopefully) able to release faster and more often. As it stands the [Mun v0.4 milestone][v4m] is a lot smaller than our original Mun v0.3 milestone.
```
